java/Makefile

```diff
@@ -86,6 +87,7 @@ else
   JAVA_RELEASE_NUMBER := $(shell echo $(JAVA_VERSION_STRING) | sed 's/\([0-9]*\)\..*/\1/')
   ifeq ($(shell expr $(JAVA_RELEASE_NUMBER) \>= 11),1)
 	DOCLET_FLAGS := --add-exports jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED
+	LINKOFFLINE := https://docs.oracle.com/en/java/javase/11/docs/api/
```

Comment:
1. I would prefer to use the latest long-term support version, which is 17.
2. Is it possible to use version 17 even when compiling under Java 8 or Java 11?  I would think so, and I think that would be nicest.

Then, this particular change (and the related changes on line 1071/1081) should be pull-requested separately, since it is a good change but has nothing to do with the main point of the pull request.

Reply:
The java17 build will fail without this change.  I seem to remember it was module related.  Java 8 doesn't have them and Java 11 does.  So I can try changing to java 17 docs, but I think it should be part of this pull request.

Reply:
changed linkoffline to jdk 17 docs